crates/bevy_ecs/src/event/mod.rs

```diff
@@ -421,30 +421,33 @@ mod tests {
     #[cfg(feature = "multi_threaded")]
     #[test]
     fn test_events_par_iter() {
-        use std::{collections::HashSet, sync::mpsc};
-
         use crate::prelude::*;
+        use std::sync::atomic::{AtomicUsize, Ordering};
+
+        #[derive(Resource)]
+        struct Counter(AtomicUsize);
 
         let mut world = World::new();
         world.init_resource::<Events<TestEvent>>();
```

Comment:
Shouldn't this be [`EventRegistry::register_event`](https://dev-docs.bevyengine.org/bevy/ecs/event/struct.EventRegistry.html#method.register_event) instead? I think just regisering the resource prevents it from existing in the registry.

Reply:
> Shouldn't this be [`EventRegistry::register_event`](https://dev-docs.bevyengine.org/bevy/ecs/event/struct.EventRegistry.html#method.register_event) instead? I think just regisering the resource prevents it from existing in the registry.

That is the more robust way to do it, but currently most existing tests use this form. I'm not sure it's worth updating all of them since they don't interact with the event registry, but in any case that should be a separate issue.

Reply:
Interestingly, world doesn't support register_event

Reply:
I *could* make an event registry from world?

Reply:
> Interestingly, world doesn't support register_event

You can do this (taken from state test):
```
EventRegistry::register_event::<StateTransitionEvent<SimpleState>>(&mut world);
```

Reply:
Should I leave this for now, and update them in another PR later? I've got three PRs waiting on Events, and am worried about merging getting complex

Reply:
But almost all existing tests don't bother with this, I don't think it's worth doing here since you didn't even add that line.